pkg/prometheus/promcfg.go

```diff
@@ -2078,3 +2084,123 @@ func (cg *ConfigGenerator) GenerateAgentConfiguration(
 
 	return yaml.Marshal(cfg)
 }
+
+func (cg *ConfigGenerator) appendScrapeConfigs(
+	slices []yaml.MapSlice,
+	scrapeConfigs map[string]*v1alpha1.ScrapeConfig,
+	apiserverConfig *v1.APIServerConfig,
+	store *assets.Store,
+	shards int32) []yaml.MapSlice {
+	scrapeConfigIdentifiers := make([]string, len(scrapeConfigs))
+	i := 0
+	for k := range scrapeConfigs {
+		scrapeConfigIdentifiers[i] = k
+		i++
+	}
+
+	// Sorting ensures, that we always generate the config in the same order.
+	sort.Strings(scrapeConfigIdentifiers)
+
+	for _, identifier := range scrapeConfigIdentifiers {
+		slices = append(slices,
+			cg.WithKeyVals("scrapeconfig", identifier).generateScrapeConfig(
+				scrapeConfigs[identifier],
+				apiserverConfig,
+				store,
+				shards,
+			),
+		)
+	}
+
+	return slices
+}
+
+func (cg *ConfigGenerator) generateScrapeConfig(
+	m *v1alpha1.ScrapeConfig,
```

Comment:
I think it might be easier/intuitive to read the code of this function if this parameter is called `sc` rather than `m`

```suggestion
	sc *v1alpha1.ScrapeConfig,
```

Reply:
Fixed in fc6aa3ef7